tsdb/engine/tsm1/cache.go

```diff
@@ -28,131 +22,75 @@ func newEntry() *entry {
 
 // add adds the given values to the entry.
 func (e *entry) add(values []Value) {
-	e.values = append(e.values, values...)
-	e.size += uint64(Values(values).Size())
-}
-
-// entries maps checkpoints to entry objects.
-type entries struct {
-	m    map[uint64]*entry
-	size uint64
-}
-
-// newEntries returns an instance of entries.
-func newEntries() *entries {
-	return &entries{
-		m: make(map[uint64]*entry),
-	}
-}
-
-// add adds checkpointed values to the entries object.
-func (a *entries) add(checkpoint uint64, values []Value) {
-	e, ok := a.m[checkpoint]
-	if !ok {
-		e = newEntry()
-		a.m[checkpoint] = e
-	}
-	oldSize := e.size
-	e.add(values)
-	a.size += e.size - oldSize
-}
-
-// evict evicts all data associated with checkpoints up to and including
-// the given checkpoint.
-func (a *entries) evict(checkpoint uint64) {
-	for k, v := range a.m {
-		if k > checkpoint {
-			continue
-		}
-		a.size -= v.size
-		delete(a.m, k)
-	}
-}
-
-func (a *entries) contains(min, max uint64) bool {
-	for k, _ := range a.m {
-		if k >= min && k <= max {
-			return true
+	// if there are existing values make sure they're all less than the first of
```

Comment:
Maybe the answer is below, but I am asking as I find it.

It seems like this code assumes the values that are being added are sorted by increasing time. If so, why is this assumption being made?